teuthology/openstack/__init__.py

```diff
@@ -191,12 +192,12 @@ def cloud_init_wait(self, name_or_ip):
                     log.debug('cloud_init_wait connect socket.error ' + str(e))
                     continue
                 except Exception as e:
-                    if 'Unknown server' not in str(e):
-                        log.exception('cloud_init_wait ' + name_or_ip)
-                    if 'Unknown server' in str(e):
-                        continue
-                    else:
-                        raise e
+                    transients = ('Incompatible ssh peer', 'Unknown server')
+                    for transient in transients:
+                        if transient in str(e):
+                            continue
+                    log.exception('cloud_init_wait ' + name_or_ip)
+                    raise e
```

Comment:
Would you please ``raise`` instead of ``raise e`` here. Re-raising the named exception obfuscates the traceback a bit.  Notice the line numbers, the ``raise e`` shows the exception happening on that line, which is not true.

```
>>> try:
...     2/0
... except Exception:
...     raise
...
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
ZeroDivisionError: integer division or modulo by zero
>>> try:
...     2/0
... except Exception as e:
...     raise e
...
Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
ZeroDivisionError: integer division or modulo by zero
```